### ISSUE TITLE
[no-useless-escape] Regexp backreference throw lint error

### ISSUE THREAD

**Issue by Delagen** [eslint/eslint#7492]
**Tell us about your environment**

* **ESLint Version:** 3.9.0
* **Node Version:** 7.0.0
* **npm Version:** 4.0.1

**What parser (default, Babel-ESLint, etc.) are you using?** gulp pipelint with manual eslint api

**Please show your full configuration:**
```
{
"rules":{
"no-useless-escape": "error"
}
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
console.log(/([\w])\1/.test("aa"));
```

**What did you expect to happen?**
No lint errors

**What actually happened? Please include the actual, raw output from ESLint.**
[no-useless-escape] Line:xx Column:xx Unnecessary escape character: \1.

**Comment by not-an-aardvark** [comment]
Thanks for the issue. However, I think this is a duplicate of https://github.com/eslint/eslint/issues/7472. It has been fixed on master, so the fix should appear in a release relatively soon.

**Comment by Delagen** [comment]
Yes it's duplicate, Thanks.
